lib/pbench/server/api/__init__.py

```diff
@@ -214,42 +225,43 @@ def post(self):
                 url = f"{elasticsearch}/{json_data['indices']}"
 
             if "payload" in json_data:
-                response = requests.post(url, json=json_data["payload"])
-            else:
-                app.logger.info(
-                    "No payload found in Elasticsearch post request json data"
+                response = requests.post(
+                    url,
+                    headers={
+                        "Content-Type": "application/json",
+                        "Accept": "application/json",
+                    },
+                    json=json_data["payload"],
                 )
-                response = requests.get(url)
+            else:
+                logger.info("No payload found in Elasticsearch request, using GET")
+                response = requests.get(url, headers={"Accept": "application/json"},)
```

Comment:
Now that I think about it we can probably have a separate definition for `get` method ?

Reply:
When we're done, the pass-through Elasticsearch and GraphQL APIs should be gone anyway in favor of native Pbench APIs that target dashboard semantics directly, so I'm not sure this is worth worrying about too much right now.

Reply:
yep and with `querycontroller` we would have a targeted API so we can build the payload inside the server, then we can have multiple such targeted endpoints and we can remove the elasticsearch endpoint from the server